src/meta.rst

```diff
@@ -71,39 +71,128 @@ A full description of selectors is
+
+Following implies that ``python`` is only a build dependency and no Python matrix will be created.
+
+.. code-block:: yaml
+
+    build:
+      - python
+    host:
+
+
+Following implies that ``python`` is a runtime dependency and a Python matrix for each supported python version will be created.
+
+.. code-block:: yaml
+
+    host:
+      - python
+
+
+
+Compilers
+---------
+
+Although ``conda-build=3`` gives the ability to use Anaconda 5 compilers, they should be used only on Windows as packages built with them are incompatible with conda-forge packages built using the CI compilers. All the packages in conda-forge needs to be rebuilt using the compilers and released simultaneously to avoid package incompatibility.
+
+Following is suggested presently,
+
+.. code-block:: yaml
+
+    requirements:
+      build:
+        - {{ compiler('cxx') }}     # [win]
+        - toolchain                 # [unix]
+
+but in the future you should switch to ``{{`compiler('cxx') }}``. An announcement will be made when this should be done.
+
+Reason for holding out using the Anaconda 5 compilers is that the packages built by them are sometimes incompatible with the packages built with the older compilers in CI platforms conda-forge has been using. All the dependencies of a package should be compiled (This is not true for all cases, but it's better to be cautious) with the new compilers before using the new compiler in a package. This presents a problem in that rebuilding a package will break the dependent packages. Therefore, ``conda-forge`` has decided to rebuild all of the packages and upload them all at once. More details on how this is done will be communicated in the future.
 
 
 Building Against NumPy
 ----------------------
 If you have a package which links\* against ``numpy`` you can build against the oldest possible version of ``numpy`` that is forwards compatible.
-That can be achieved by pinning the build requirements and letting "free" the run requirements.
+That can be achieved by pinning the host requirements and letting "free" the run requirements.
 If you don't want to make things complicated you can use
 
 .. code-block:: yaml
 
-    build:
-      - numpy 1.11.*
+    host:
+      - numpy
     run:
-      - numpy >=1.11
+      - {{ pin_compatible('numpy', min_pin='x.x') }}
 
-However, these are the oldest available ``numpy`` versions in conda-forge that you can use, if you need to support older versions than ``numpy 1.11``.
+
+At the time of writing, above is equivalent to the following,
 
 .. code-block:: yaml
 
-    build:
+    host:
+      - numpy 1.9.3              # [unix]
+      - numpy 1.11.3             # [win]
+    run:
+      - numpy >=1.9.3,<2.0.a0    # [unix]
+      - numpy >=1.11.3,<2.0.a0   # [win]
+
+
+However, these are not the oldest available ``numpy`` versions in conda-forge that you can use, if you need to support older versions,
```

Comment:
If you wanted these old versions, I'd recommend changing conda-forge-pinning to have equivalent selectors.  Try to keep the pattern of pinning in recipes to as little as possible, and only when the pins in conda-forge-pinning just won't work for some reason.  One reason why they won't work sometimes is because you need a newer base version than the base version in the config.  For example, scikit-learn requires numpy 1.11, so I would put that in scikit-learn's numpy host requirement, because it is newer than the base versions for numpy in conda_build_config.yaml

Reply:
cc @ocefpaf. Personally I don't care about numpy before 1.11. I think we can change this example for requiring newer numpy versions.